refactor: Kotlinify symbol keyboard

nopdan opened the pull request:
## Pull request

液态键盘相关

#### Issue tracker

Fixes will automatically close the related issues

#### Feature
Describe features of pull request

#### Code of conduct
- [x] [CONTRIBUTING](CONTRIBUTING.md)

#### Style lint
- [x] `make sytle-lint`

#### Build pass
- [x] `make debug`

#### Manually test
- [x] Done

#### Code Review
1. No wildcards import
2. Manual build and test pass
3. GitHub action ci pass
4. At least one contributor reviews and votes
5. Can be merged clean without conflicts
6. PR will be merged by rebase upstream base

#### Daily build
Login and download artifact at https://github.com/osfans/trime/actions

#### Additional Info

nopdan committed:
refactor: Kotlinify symbol

WhiredPlanck commented on app/src/main/java/com/osfans/trime/ime/symbol/SimpleKeyBean.kt:
```diff
@@ -0,0 +1,25 @@
+
+import java.io.Serializable
+
+class SimpleKeyBean : Serializable {
```
这个可以改成 data class 以简化代码

WhiredPlanck commented on app/src/main/java/com/osfans/trime/ime/symbol/TabManager.kt:
```diff
@@ -0,0 +1,231 @@
+import timber.log.Timber
+
+// 使用TabManager时，不应该使用变量保存TabManager实例，应该使用TabManager.get()方法获取
+class TabManager private constructor() {
```
这个弄成单例类（object）可能会更好

WhiredPlanck commented on app/src/main/java/com/osfans/trime/ime/symbol/TabTag.kt:
```diff
@@ -0,0 +1,36 @@
+
+// Tab是滑动键盘顶部的标签按钮（包含返回键）。
+// 为了公用候选栏的皮肤参数以及外观，保持了和普通键盘布局相似的代码。此类相当于原键盘布局的Rime.RimeCandidate
+class TabTag {
```
这个改为 data class 可以简化代码

WhiredPlanck committed:
refactor(symbol): use data class to make life easier

WhiredPlanck committed:
refactor(symbol): increase adapters' code quality

WhiredPlanck committed:
refactor(symbol): singleton TabManager

WhiredPlanck committed:
fix: random crash on select LiquidKeyboard tab

RecycleView's notifyItemRangeChanged may has bugs, use notifyItemRangeRemoved and notifyItemRangeInserted.
